examples/basic/gpu_shared_memory.c

```diff
@@ -0,0 +1,245 @@
+/*
```

Comment:
1. Is this fille in the correct directory?
2. Please keep comparisons consistent, once in code you do ```NULL==VARIABLE``` in other places you do ```VARIABLE == NULL``` (i vote for second one)
3. nit: consider to change error handling from 
``` c
int fun1() {
    void *ptr1, *ptr2, *ptr3;
    ptr1 = malloc(10);
    if (!ptr1) {
        return 1;
    }
    ptr2 = malloc(10);
    if (!ptr2) {
        goto free1;
    }
    ptr3 = malloc(10);
    if (!ptr3) {
        goto free2;
    }
    free(ptr3);
    free(ptr2);
    free(ptr1);
    return 0;
free2:
    free(ptr2);
free1:
    free(ptr1);
    return 1;
}
``` 
to
``` c
int fun2() {
    void *ptr1, *ptr2, *ptr3;
    int ret = 0;
    ptr1 = malloc(10);
    if (!ptr1) {
        ret = 1;
        goto e1;
    }
    ptr2 = malloc(10);
    if (!ptr2) {
        ret = 1;
        goto e2;
    }
    ptr3 = malloc(10);
    if (!ptr3) {
        ret = 1;
        goto e3;
    }
    free(ptr3);
e3:
    free(ptr2);
e2:
    free(ptr1);
e1:
    return ret;
}

```